R/to_factor.R

```diff
@@ -125,3 +118,34 @@ to_factor.data.frame <- function(x,
   x[select] <- lapply(x[select], to_factor, verbose = verbose, ...)
   x
 }
+
+
+
+# helper -----------------------
+
+.value_labels_to_levels <- function(x, remove_attr = TRUE, verbose = TRUE) {
+  # extract value labels
+  value_labels <- attr(x, "labels", exact = TRUE)
+  # return, if none
+  if (is.null(value_labels)) {
+    return(x)
+  }
+  # check positions of matching values and levels
+  levels_in_labs <- stats::na.omit(match(value_labels, levels(x)))
+  labs_in_levels <- stats::na.omit(match(levels(x), value_labels))
+  # sanity check - if labelled values and levels don't match
+  if (!length(levels_in_labs) || !length(labs_in_levels)) {
+    if (verbose) {
+      insight::format_alert(
+        "Could not use value labels as factor levels.",
+        "Labelled values and factor levels didn't macth."
+      )
```

Comment:
"Labelled values and factor levels didn't macth." -> "Labelled values and factor levels had no match."

Also "macth" -> "match"